pallets/elections/src/lib.rs

```diff
@@ -311,15 +311,27 @@ pub mod pallet {
         type Error = ElectionError;
         type DataProvider = T::DataProvider;
 
-        // The elections are PoA so only the nodes listed in the Validators will be elected as validators.
-        // We calculate the supports for them for the sake of eras payouts.
+        /// The elections are PoA so only the nodes listed in the Validators will be elected as
+        /// validators. It applies except `NextEraReservedValidators`, which are chosen no matter
+        /// their stake nor support.
+        ///
+        /// We calculate the supports for them for the sake of eras payouts.
         fn elect() -> Result<Supports<T::AccountId>, Self::Error> {
-            let voters =
-                Self::DataProvider::electing_voters(None).map_err(Self::Error::DataProvider)?;
-            let validators = NextEraReservedValidators::<T>::get()
+            let staking_validators = Self::DataProvider::electable_targets(None)
+                .map_err(Self::Error::DataProvider)?
+                .into_iter()
+                .collect::<BTreeSet<_>>();
+            let reserved_validators: BTreeSet<_> = NextEraReservedValidators::<T>::get()
+                .into_iter()
+                .collect::<BTreeSet<_>>();
+            let non_reserved_validators: BTreeSet<_> = NextEraNonReservedValidators::<T>::get()
+                .into_iter()
+                .collect::<BTreeSet<_>>();
+
+            let eligible_validators =
+                &(&staking_validators & &non_reserved_validators) | &reserved_validators;
```

Comment:
Hilarious, this looks almost like python!

Reply:
and it is a way more handy than `intersection()` / `union()` methods that return intermediate lazy structs, which you have to `iter().collect()` to perform another operation

Reply:
Those `&` before values obfuscate the whole expression a bit but agree `&` and `|` are handy.